### PR TITLE
Remove deprecated logging property - Fix cancellation token usage in AMQP

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs [hunk 1]
```diff
@@ -92,7 +91,7 @@ private async Task InitializeAndSetupClientAsync(CancellationToken cancellationT
             if (ShouldClientBeInitialized(s_connectionStatus))
             {
                 // Allow a single thread to dispose and initialize the client instance.
-                await _initSemaphore.WaitAsync();
+                await _initSemaphore.WaitAsync(cancellationToken);
```
Comment (drwill-ms) [inline iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs:94]: May need to handle an exception that is thrown if this is cancelled.

Comment (drwill-ms) [reply iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs:94]: Same with the calls below.

File: iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/Program.cs [hunk 2]
```diff
@@ -77,7 +77,7 @@ private static ILogger InitializeConsoleDebugLogger()
             {
                 builder
                 .AddFilter(level => level >= LogLevel.Debug)
-                .AddConsole(options =>
+                .AddSystemdConsole(options =>
```
Comment (drwill-ms) [inline iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/Program.cs:80]: Can you say more about this change?

Comment (azabbasi) [reply iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/Program.cs:80]: https://docs.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/obsolete-consoleloggeroptions-properties

ConsoleLoggerOptions properties have been deprecated, it's recommended to use ConsoleFormatterOptions properties instead.

Comment (azabbasi) [reply iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/Program.cs:80]: This was the only warning that started popping up on the Device samples that I had to fix. back to 0.

File: iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs (hunk 1 above)
Comment (azabbasi) [reply iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs:94]: I think it's best if catch that exception at the program entry and log it. let me what you think.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**